packages/assets-controllers/src/AssetsContractController.ts

```diff
@@ -207,10 +300,24 @@ export class AssetsContractController extends BaseControllerV1<
    * @param networkClientId - Network Client ID used to get the provider.
    * @returns Hex chain ID.
    */
-  getChainId(networkClientId?: NetworkClientId): Hex {
-    return networkClientId
-      ? this.getNetworkClientById(networkClientId).configuration.chainId
-      : this.config.chainId;
+  #getCorrectChainId(networkClientId?: NetworkClientId): Hex {
+    if (networkClientId) {
+      const networkClientConfiguration = this.messagingSystem.call(
+        'NetworkController:getNetworkConfigurationByNetworkClientId',
+        networkClientId,
+      );
+      if (networkClientConfiguration) {
+        return networkClientConfiguration.chainId;
+      }
+    }
+    const { selectedNetworkClientId } = this.messagingSystem.call(
+      'NetworkController:getState',
+    );
+    const networkClient = this.messagingSystem.call(
+      'NetworkController:getNetworkClientById',
+      selectedNetworkClientId,
+    );
+    return networkClient.configuration?.chainId ?? this.#chainId;
```

Comment:
Retrieves the currently selected chainId even if the `networkClientId` argument is not provided (fixes breaking tests).